dags/iop/parser.py

```diff
@@ -49,10 +54,25 @@ def _get_journal_doctype(self, article: ET.Element):
             self.logger.error("Article-type is not found in XML", doi=self.dois)
             return None
         try:
-            return self.article_type_mapping[value]
+            self.journal_doctype = self.article_type_mapping[value]
+            return self.journal_doctype
         except KeyError:
             self.logger.error(
                 "Unmapped article type", doi=self.dois, article_type=value
             )
         except Exception:
             self.logger.error("Unknown error", doi=self.dois)
+
+    def _get_related_article_doi(self, article):
+        try:
+            if self.journal_doctype in ["corrigendum", "addendum"]:
+                node = article.find(
+                    "front/article-meta/related-article[@ext-link-type='doi']"
+                )
+                related_article_doi = node.get("href")
+                if related_article_doi:
+                    self.logger.info("Adding related_article_doi.")
+                    return related_article_doi
+        except AttributeError:
+            self.logger.error("No related article dois found", doi=self.dois)
+            return None
```

Comment:
```suggestion
    def _get_related_article_doi(self, article):
        try:
            if self.journal_doctype not in ["corrigendum", "addendum"]:
                return
            node = article.find(
                "front/article-meta/related-article[@ext-link-type='doi']"
            )
            related_article_doi = node.get("href")
            if not related_article_doi:
                return
            self.logger.info("Adding related_article_doi.", doi=self.doi)
            return related_article_doi
        except AttributeError:
            self.logger.error("No related article dois found", doi=self.dois)
            return
```

Reply:
Also we shouldn't wrap everything in the try-except, what could throw an `AttributeError`?

Reply:
no, in both possible error cases could be an attribute error:
`article.find`
`node.get`

Reply:
then I would split into 2 try/except, the problem with the current now is you don't know which failed.

Reply:
if `find` doesn't throw an `AttributeError` why to include it then? :)

Reply:
It is not included 😅: https://github.com/cern-sis/workflows/blob/2874f7d835d755f20707f51ebac092cd4a9eebb2/dags/iop/parser.py#L69

Reply:
Before I was catching the error which should bever happen: https://github.com/cern-sis/workflows/pull/67#discussion_r1045927102

But later I realized, that actually one is enough. 
Did I understand the comment correct?

Reply:
you changed it correclty 👍